### PR TITLE
Hitscan Emitters 2 Electric Boogaloo

### PR DESCRIPTION
**Pull request by SynnGraffkin**
## About The Pull Request

Ports [(is this really a good idea) hitscan emitters #10450](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10450)

Emitters are now hitscan.
raises hitscan safety to 10x range, since there's already a safety mechanism for reflections.

## Why It's Good For The Game

Haha, Emitters go PEW.

image

## Changelog
:cl: silicons
balance: emitters are now hitscan
/:cl:

### CODE REVIEW

**Comment by MarkSuckerberg** [comment]
oh hey this fixes the bug we were talking about in #204 where the emitters and guns were fucky

**Comment by MarkSuckerberg** [comment]
also fuck

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -123,6 +123,15 @@
...
 
+//Wasp Edit - Hitscan Emitters
+/obj/projectile/beam/emitter/hitscan
+	hitscan = TRUE
+	tracer_type = /obj/effect/projectile/tracer/laser/emitter
+	muzzle_type = /obj/effect/projectile/muzzle/laser/emitter
+	impact_type = /obj/effect/projectile/impact/laser/emitter
+	impact_effect_type = null
+//Wasp End
+
```
Comment (Superyodeler) [inline code/modules/projectiles/projectile/beams.dm:134]: This

File: code/game/objects/effects/temporary_visuals/projectiles/impact.dm [hunk 2]
```diff
@@ -36,3 +36,8 @@
 
 /obj/effect/projectile/impact/wormhole
 	icon_state = "wormhole_g"
+
+//Wasp - Hitscan Emitters
+/obj/effect/projectile/impact/laser/emitter
+	name = "emitter impact"
+	icon_state = "impact_emitter"
```
Comment (Superyodeler) [inline code/game/objects/effects/temporary_visuals/projectiles/impact.dm:43]: This

File: code/game/objects/effects/temporary_visuals/projectiles/muzzle.dm [hunk 3]
```diff
@@ -28,3 +28,8 @@
 
 /obj/effect/projectile/muzzle/wormhole
 	icon_state = "wormhole_g"
+
+//Wasp - Hitscan Emitters
+/obj/effect/projectile/muzzle/laser/emitter
+	name = "emitter flash"
+	icon_state = "muzzle_emitter"
```
Comment (Superyodeler) [inline code/game/objects/effects/temporary_visuals/projectiles/muzzle.dm:35]: This

File: code/game/objects/effects/temporary_visuals/projectiles/tracer.dm [hunk 4]
```diff
@@ -66,3 +66,8 @@
 
 /obj/effect/projectile/tracer/wormhole
 	icon_state = "wormhole_g"
+
+//Wasp - Hitscan Emitters
+/obj/effect/projectile/tracer/laser/emitter
+	name = "emitter beam"
+	icon_state = "emitter"
```
Comment (Superyodeler) [inline code/game/objects/effects/temporary_visuals/projectiles/tracer.dm:73]: This

**Comment by SynnGraffkin** [comment]
reeeeeeeeeeeeeeeeeeeeeeeee

**Comment by SynnGraffkin** [comment]
@Superyodeler Does it really need to be modular though?

**Comment by Superyodeler** [comment]
It'll mean less work marging tg prs in the future, and it shouldn't take TOO long to set up.

**Comment by SynnGraffkin** [comment]
and makes a lot of extra dm files to compile for a small addition

**Comment by Maldaris** [comment]
> and makes a lot of extra dm files to compile for a small addition

Extra DM files with small additions don't actually take that long to compile. Not having to resolve merge conflicts down the line is far more valuable.

**Comment by SynnGraffkin** [comment]
I really don't want to make a bunch of small dm files just for one child of a laser.

**Comment by Maldaris** [comment]
Nothing says you can't just put em all in the same file. See spider margularization

**Comment by SynnGraffkin** [comment]
@MarkSuckerberg is the one primarily touching most of the files with TG updates at this point, so let's ask him.

**Comment by Maldaris** [comment]
Margularized it since Toast was dragging his heels on this one. :eyes:

File: code/game/objects/effects/temporary_visuals/projectiles/tracer.dm (hunk 4 above)
Comment (Maldaris) [reply code/game/objects/effects/temporary_visuals/projectiles/tracer.dm]: resolved in 80650be07c

File: code/game/objects/effects/temporary_visuals/projectiles/muzzle.dm (hunk 3 above)
Comment (Maldaris) [reply code/game/objects/effects/temporary_visuals/projectiles/muzzle.dm]: resolved in 80650be07c

File: code/game/objects/effects/temporary_visuals/projectiles/impact.dm (hunk 2 above)
Comment (Maldaris) [reply code/game/objects/effects/temporary_visuals/projectiles/impact.dm]: resolved in 80650be07c

File: code/modules/projectiles/projectile/beams.dm (hunk 1 above)
Comment (Maldaris) [reply code/modules/projectiles/projectile/beams.dm]: resolved in 80650be07c

**Comment by SynnGraffkin** [comment]
You perverted my PR with your hands and somehow desecrated the whole DME

**Pull request merged**